src/formatter.rs

```diff
@@ -156,36 +162,35 @@ impl<W: Write> ColorFormatter<W> {
         if let Some(cached) = self.cached_styles.get(&self.labels) {
             cached.clone()
         } else {
-            let mut best_match = (-1, Style::default());
+            // We use the reverse list of matched indices as a measure of how well the rule
+            // matches the actual labels. For example, for rule "a c" and the actual labels
+            // "a b c", we'll get [2,0]. We compare them by Rust's default vector
+            // comparison. That means "a c" will trump both rule "b" (priority [1]) and rule
+            // "a b" (priority [1,0]).
+            let mut matched_styles = vec![];
```

Comment:
Nit: it would be nice if "c" (priority [2]) is included in the example. "a c" vs "a b" only requires the maximum index, but "a c" vs "c" doesn't.

Reply:
Good point. I think I meant something different than I wrote. I've update the example to be more complex and hopefully more useful.